pkg/services/authz/zanzana/store/sqlite/write.go

```diff
@@ -0,0 +1,165 @@
+package sqlite
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/oklog/ulid/v2"
+	"google.golang.org/protobuf/proto"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"github.com/openfga/openfga/pkg/storage"
+	tupleUtils "github.com/openfga/openfga/pkg/tuple"
+)
+
+// write is copied from https://github.com/openfga/openfga/blob/main/pkg/storage/sqlcommon/sqlcommon.go#L330-L456
+// but uses custom handleSQLError.
+func write(
```

Comment:
@DanCech I had to copy think function in order for us to use a custom handleSQLError function. Otherwise the test suite for stores provided by open fga will fail because our wrapper will never be called. I will create an issue upstream to see if we can resolve that so we don't have to maintain this copy

Reply:
Yeah, it seems like that's one area where the store needs to get the first bite at the error instead of code for every store being in one central place

Reply:
Yeah I created an issue and proposed the ability to pass the error handler as a parameter to common database operations such as this one https://github.com/openfga/openfga/issues/1709

Reply:
Another option would be to move calling handleSqlError from within sqlcommon out into Write in the store itself, which might be simpler and cleaner since write doesn't actually use the handled error

Reply:
True, that would be much simpler